.github/workflows/publish.yml

```
@@ -25,12 +25,17 @@ jobs:
         with:
           node-version-file: '.nvmrc'
           registry-url: 'https://registry.npmjs.org'
-      - run: npm ci
-      - run: echo "pkg_name=${GITHUB_REF_NAME##*@}" >> "$GITHUB_ENV"
+        - run: echo "pkg_name=${GITHUB_REF_NAME##*@}" >> "$GITHUB_ENV"
       - run: echo "pkg_version=${GITHUB_REF_NAME%@*}" >> "$GITHUB_ENV"
+      - run: npm ci
+      - name: check integretiy between regestry and package.json
+        run: node --run pjson:check
       - name: Generate types
         run: node --run types:generate
-      - name: Publish $pkg_name $pkg_version with provenance
+    # try first publish to npm because it's our primary registry
+      - name: Publish $pkg_name $pkg_version with provenance to npm
```

nit: easier to see on a small screen (better to show npm vs jsr first)

```suggestion
      - name: Publish $pkg_name $pkg_version to nom with provenance
```